### DISCUSSION TITLE
Sending Mail via SMTP fails, because of "Caused by: jakarta.mail.MessagingException: 501 Syntax: HELO <hostname>"

### DISCUSSION THREAD

**Discussion by TommiCodes** [keycloak/keycloak#30671] [Q&A] [answered]
We want to send Emails with keycloak and therefore we are trying to set it up with smtp.
When we try to send a testmail it always fails with: "Caused by: jakarta.mail.MessagingException: 501 Syntax: HELO <hostname>".

I also tried to set a hostname for the mailer via ENV JAVA_OPTS_APPEND="-Dmail.smtp.localhost=myapp.local" and so on.

In Keycloak the "KC_HOSTNAME" is set to our domain, so xxx.com

For the connection in keycloak we are using:
Host: email-smtp.eu-central-1.amazonaws.com
Port: 465
SSL & StartTls ENABLED
Authentication with correct username and password

Keycloak itself is running fine in https Mode and has a certificate.
Some words to our environment:
- We are using Keycloak: 25
- We deploy to AWS ECS via Terraform
- We are using AWS SES for sending Mails (the Testmails from SES itself are working)
- Our Domain points to a loadbalancer and there the request is forwarded to our target group

Has anyone an idea what i am doing wrong or has a hint? Thanks :)

**Comment by SuadHus** [comment]
If i am not mistaken, i think you gotta have either ssl or startTls enabled not both, try inactivating StartTls.

**Reply by TommiCodes** [reply]
I tried all combinations, just ssl or just startTls, both and so on, but neither is working.
Always also updated the port according to aws ses docs

**Comment by BrickmanRodrigo** [comment]
We had the same error in 24.0.4, we realized that the /etc/hosts file was malformed

`127.0.0.1 localhost`
`x.x.x.x`

x.x.x.x being the private address of the container in AWS ECS Fargate. Manually changing this to just the first line makes the problem go away.
We also have the KC_HOSTNAME environment variable setup correctly

Let us know if you find anything else!

**Comment by TommiCodes** [comment, marked as answer]
Hi Rodrigo, thanks for the heads up.
I noticed the same and wrote a script for aws ecs fargate to update the localhost.

Before executing the script i installed the dependencies in a builder step in my dockerfile and copied the files to the next stage.

Then the script is being executed on startup of my fargate task.
This is the relevant part of the script:
```
# Get the container's IP address and hostname
container_ip=$(ip a | grep -A2 eth0 | grep inet | awk '{print $2}' | sed 's#/.*##g')
container_hostname=$(hostname)

# Update /etc/hosts with the container's IP and hostname
echo "$container_ip $container_hostname" >> /etc/hosts
```

I think both solutions will work fine.

FR

**Reply by theophanevie** [reply]
Worked well for me, thanks for sharing.
